src/Microsoft.AspNetCore.Routing/Tree/TreeRouter.cs

```diff
@@ -50,6 +56,39 @@ public class TreeRouter : IRouter
             ILogger routeLogger,
             ILogger constraintLogger,
             int version)
+            : this(trees, linkGenerationEntries, objectPool, routeLogger, constraintLogger, version, urlEncoder)
```

Comment:
All comments made in `TemplateBinder` apply to this class as well. Don't need more than one new constructor.